feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/NftListInteractor.kt

```diff
@@ -62,23 +78,20 @@ class NftListInteractor(
 
     @OptIn(ExperimentalCoroutinesApi::class)
     fun subscribeNftOwnerChanged(): Flow<Unit> {
-        return nftRepository.getPendingSendTransactionsNftIds()
-            .flatMapLatest(::subscribeNftsOwnerAddresses)
-            .onEach {
-                val blockTimeInMillis = chainStateRepository.expectedBlockTimeInMillis(it.chainId)
-                delay(blockTimeInMillis.toLong())
-                syncNftsList()
-            }.map {}
+        return pendingSendNftTransactionRepository.getPendingSendTransactionsNftIds()
+            .flatMapLatest(::subscribeNftsOwnerAccountId)
+            .onEach { syncNftListFromNextBlock(it.chainId) }
+            .map {}
     }
 
-    private suspend fun subscribeNftsOwnerAddresses(nftIds: Set<String>): Flow<NftLocal> {
-        val myAccountAddress = accountRepository.getSelectedMetaAccount().substrateAccountId?.toHexString()
+    private suspend fun subscribeNftsOwnerAccountId(nftIds: Set<String>): Flow<NftLocal> {
         return nftIds.map { nftId ->
-            val nftLocal = nftRepository.getLocalNft(nftId)
-            nftRepository.subscribeNftOwnerAddress(nftLocal)
+            val chain = nftRepository.getChainForNftId(nftId)
+            val myAccountAddress = accountRepository.getSelectedMetaAccount().accountIdIn(chain)
+            nftRepository.subscribeNftOwnerAccountId(nftId)
                 .distinctUntilChanged()
-                .filter { ownerAddress -> myAccountAddress != ownerAddress }
-                .map { nftLocal }
+                .filter { (ownerAddress, _) -> !(myAccountAddress contentEquals ownerAddress) }
+                .map { (_, nftLocal) -> nftLocal }
```

Comment:
Seems like we're still making a subscription for each nft. To achieve bulk subscription, you should reuse single `subscriptionBuilder` cross multiple nfts. It wokrs in a way that `subscriptionBuilder.subscribe(key)` adds a subscription to the pending list and once `subcribe(scope)` is called all pending subscriptions are sent in a single request to the node

Reply:
Done